presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -143,6 +151,25 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
         return planNodeOrderings.min(possibleJoinNodes).getPlanNode();
     }
 
+    public static JoinNode confidenceBasedBroadcast(JoinNode joinNode, Context context)
+    {
+        if (!mustPartition(joinNode)) {
+            if (confidenceBasedBroadcastingEnabled(context.getSession())) {
+                boolean rightIsAtleastHighConfidence = isAtleastHighConfidence(joinNode.getRight(), context);
```

Comment:
I was thinking that, FACT > LOW/HIGH and HIGH > LOW, not just at least HIGH

Reply:
@feilong-liu I thought about this as well. But I was thinking if there was something with High Confidence with a lower estimate than something with Fact Confidence, wouldn't it make sense to pick the smaller one?

Reply:
High confidence can still be inaccurate, fact is always true

Reply:
Okay that makes sense, I'll go ahead and make that change